src/Nox.Lib/Middlewares/RelatedEndpointsMiddleware.cs

```diff
@@ -166,4 +176,19 @@ private PathString BuildNewPatchPath(List<string> segments)
             return new PathString(_apiPrefix + "/" + _navigationNameToEntityPluralName[segments[count - 2]] + "/" + segments[^1]);
         }
     }
+
+    private PathString BuildNewPath(List<string> segments)
+    {
+        int count = segments.Count;
+        bool isEvenCount = count % 2 != 0;
+
+        if (isEvenCount)
+        {
+            return new PathString(_apiPrefix + "/" + _navigationNameToEntityPluralName[segments[count - 3]] + "/" + segments[count - 2] + "/" + segments[^1]);
```

Comment:
@anna-naumova25  I would suggest to have the _apiPrefix + "/" precomputed in some variable (or add the / to the _apiPrefix when its read from the configuration), avoiding creating an extra string every time...

Reply:
have updated _apiPrefix